aten/src/ATen/native/Integration.cpp

```diff
@@ -52,7 +52,21 @@ Tensor do_cumulative_trapezoid(const Tensor& y, double dx, int64_t dim) {
 
     return (dx /2. * (left + right)).cumsum(dim);
 }
-
+// Given the current shape of a Tensor and a target number of dimensions,
```

Comment:
Why not just use expand for this?

https://pytorch.org/docs/master/generated/torch.Tensor.expand.html#torch.Tensor.expand

Reply:
The behavior that we want (and `numpy` has) is for `x` and `y` tensors to broadcast together.

If we do `x.expand(y.sizes())`, we would be broadcasting `x` into the shape of `y`. Therefore, in this part, I create a custom function to add padding of `1`s to the shape of `x` when it is has a smaller dimension than `y`.

Reply:
They could be expanded to the common size, though. There's also broadcast_tensors():

https://pytorch.org/docs/master/generated/torch.broadcast_tensors.html?highlight=broadcast_tensors#torch.broadcast_tensors

It'd be weird if we didn't have an existing function for this

Reply:
Oh one more thing, `numpy` also broadcast after taking the difference of each tensor. If we broadcast here, it would be before we take the difference of the tensors.

Reply:
You mean this is still divergent with NumPy and we should broadcast later? Or the current behavior handles this caveat?

Reply:
The current behavior (with the custom padding function) does the bare minimum to increase `x`'s number of dimension if needed. Like NumPy, the broadcasting of two tensors happens **at a later time after** the differences are taken.

I believe if we call something like `x.expand(y.sizes())` or `x.expand(broadcast_tensors(x,y)[0])` here, instead of using the padding function, we would be broadcasting **prior** to the differences being taken and therefore it will diverge from NumPy.

Reply:
That could be. Let's just clearly document this concern / rationale then so people reading this code in the future (like us 1-2 years from now) will remember why we did it

Reply:
The current one just got merged. I am going to add another commit to make sure the comment is there.